### PR TITLE
MRF / JS Factory function completed 🥇 

### PR DESCRIPTION
**Pull request by Mr-MRF-Dev**
related: #37

### CODE REVIEW

File: MRF/Js/js-37/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+//using factory function
+function callAPI({ baseURL }) {
+  return {
+    baseURL: baseURL,
+    get: function () {
+      return "get method called";
```
Comment (mkermani144) [inline MRF/Js/js-37/index.js:6]: These two functions are meant to call `fetch`, like ```fetch(`${baseURL}/${endpoint}`)```.

File: MRF/Js/js-37/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+//using factory function
+function callAPI({ baseURL }) {
+  return {
+    baseURL: baseURL,
```
Comment (mkermani144) [inline MRF/Js/js-37/index.js:4]: This is not needed.

**Review by mkermani144 — CHANGES_REQUESTED** [review]

File: MRF/Js/js-37/index.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
+//using factory function
+function callAPI({ baseURL, endPoint }) {
```
Comment (mkermani144) [inline MRF/Js/js-37/index.js:2]: It doesn't make sense to put endPoint here. With this implementation, if someone wants to `get` endpoint `x` and `post` endpoint `y` for the same base url, he/she should call the factory function twice, which totally negates the benefits of factory functions. Same applies to the class implementation.  
You always need to ask which configs or parameters are set once, and which ones need to be set per use case. In an endpoint fetching scenario, there is a high chance you never alter the base url, but request different endpoints based on use case.

**Review by mkermani144 — CHANGES_REQUESTED** [review]

File: MRF/Js/js-37/index.js (hunk 3 above)
Comment (Mr-MRF-Dev) [reply MRF/Js/js-37/index.js:2]: Yes, I had also considered this. I believed creating an object for each endpoint would be more efficient than specifying the endpoint each time. after refactoring the code I think your approach is better.

**Review by mkermani144 — CHANGES_REQUESTED** [review]
Please sync your branch with main.

**Review by mkermani144 — APPROVED** [review]

**Pull request merged**